### ISSUE TITLE
Comparisons to other libraries

### ISSUE THREAD

**Issue by 0x8890** [monkberry/monkberry#18]
I'm curious to hear how this is different from other libraries, such as [HTMLBars](https://github.com/tildeio/htmlbars).

**Comment by elfet** [comment]
Monkberry is really different from HTMLBars, it's not string template engine, it's DOM template engine. 
It's more like React or Angular.

**Comment by elfet** [comment]
Another big difference is what Monkberry does not use `eval` or it's analog `new Function` to create template. 

From [HTMLBars source](https://github.com/tildeio/htmlbars/blob/master/packages/htmlbars-compiler/lib/compiler.js#L35):
```js
export function template(templateSpec) {
  return new Function("return " + templateSpec)();
}
```

**Comment by elfet** [comment]
Size matter to:
```
$ jsize htmlbars monkberry
htmlbars 	 41.84 kB (gzip)
monkberry 	 1.16 kB (gzip)
```

**Comment by 0x8890** [comment]
Hmm, I'm not sure if that is the case, the [runtime](https://github.com/tildeio/htmlbars/tree/master/packages/htmlbars-runtime/lib) is quite small.

Also, the compiled template looks to implement a similar concept to what monkberry does, build dom code from a template string, especially the `build` function: https://github.com/emberjs/ember.js/issues/10454

**Comment by uhjish** [comment]
On the same note, how does this compare to [sveltejs](https://svelte.technology/guide)? Both seem to compile templates down into optimized vanillaJS and html... Is the compiled code generated by monkberry stand-alone or does it require a runtime?

**Comment by leeoniya** [comment]
from http://monkberry.js.org/docs/installation it does require a runtime. but the runtime is 1.25k min/gz, so pretty much negligible. the size of the generated code would be a much better comparison though.

**Comment by uhjish** [comment]
makes sense, and the --as-module compile leads me to believe that it can build self-contained ES6 modules, which could again be composed into optimized apps with a bit of tree-shaking ala rollup or webpack2... neat.

**Comment by elfet** [comment]
@uhjish sveltejs is pretty awesome, but monkberry was first 😃

**Comment by leeoniya** [comment]
and sveltejs got to these first:

https://github.com/monkberry/monkberry/issues/35 each-else: https://github.com/sveltejs/svelte/pull/122
https://github.com/monkberry/monkberry/issues/37 keyed updates: https://github.com/sveltejs/svelte/pull/209
https://github.com/monkberry/monkberry/issues/2 SSR: https://github.com/sveltejs/svelte/issues/196

...maybe others

at the end of the day what matters is what a lib offers *today*, not who was first to an MVP [1]. for what it's worth, i consider sveltejs to also be an mvp due to how many use cases seem to require some form of "well yeah" work-arounds :P

[1] https://en.wikipedia.org/wiki/Minimum_viable_product

**Comment by uhjish** [comment]
@elfet Imitation is the sincerest form of flattery; forking is a close second ;-).

**Comment by elfet** [comment]
@leeoniya svelte is awesome and have much more "media" power, better marketing. :) 

I have plans to add keyed updates in future, btw.
